dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/IdentifiableObjectUtils.java

```diff
@@ -396,4 +395,27 @@ else if ( identifiableObject.getCode() != null && !identifiableObject.getCode().
 
         return object.getClass().getName();
     }
+
+    public static <T extends BaseIdentifiableObject> String getIdentifierBasedOnIdScheme( T object, IdScheme idScheme )
+    {
+        //idScheme with IdentifiableProperty.ATTRIBUTE has to be specially treated in the code dealing with Attribute IDs
```

Comment:
Is it possible to move that logic into this method?

Reply:
Not really. The logic will differ every time. The Attribute:UID has to be treated in a specific way. Right now I am not able to obtain Attribute:UID from DE UID. Also, the method is generic for all IdentifiableObjects. But if we figure out how to do a mapping for Attribute:UIDs, then sure, it should be moved here.